### PR TITLE
[Az.RecoveryServices.Backup] Added support for non-UTC timezones with standard policies

### PR DESCRIPTION
**Pull request by hiaga**
## Description

Added support for non-UTC timezones with standard policy for workloadType IaasVM, MSSql, AzureFiles.

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/RecoveryServices/RecoveryServices.Backup.Helpers/Conversions/ConversionHelpers.cs [hunk 1]
```diff
@@ -856,9 +876,10 @@ public static List<ItemBase> GetItemModelListCrr(IEnumerable<CrrModel.ProtectedI
                 }
                 else if (string.Compare(subProtectionPolicy.PolicyType, "Log") == 0)
                 {
+                    // timeZone paased as input but not used in below method calls
                     azureVmWorkloadPolicyModel.LogBackupSchedulePolicy = PolicyHelpers.GetPSLogSchedulePolicy((ServiceClientModel.LogSchedulePolicy)
                     subProtectionPolicy.SchedulePolicy,
-                    ((ServiceClientModel.AzureVmWorkloadProtectionPolicy)serviceClientResponse.Properties).Settings.TimeZone);
+                    ((ServiceClientModel.AzureVmWorkloadProtectionPolicy)serviceClientResponse.Properties).Settings.TimeZone);
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices.Backup.Helpers/Conversions/ConversionHelpers.cs:882]:
```suggestion
                    ((ServiceClientModel.AzureVmWorkloadProtectionPolicy)serviceClientResponse.Properties).Settings.TimeZone);
```

File: src/RecoveryServices/RecoveryServices.Backup.Helpers/Conversions/SchedulePolicyConversions.cs [hunk 2]
```diff
@@ -149,13 +149,13 @@ public partial class PolicyHelpers
                 
                 // multiple backups per day 
                 psPolicy.HourlySchedule.Interval = serviceClientPolicy.HourlySchedule.Interval;
-                psPolicy.HourlySchedule.WindowStartTime = serviceClientPolicy.HourlySchedule.ScheduleWindowStartTime;
+                psPolicy.HourlySchedule.WindowStartTime = serviceClientPolicy.HourlySchedule.ScheduleWindowStartTime; // checkUTC: do we need to convert here too ?
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices.Backup.Helpers/Conversions/SchedulePolicyConversions.cs:152]: Please do not leave a question here.

File: src/RecoveryServices/RecoveryServices.Backup.Models/CommonModels/PolicyScheduleObjects.cs [hunk 3]
```diff
@@ -116,12 +116,12 @@ public override void Validate()
                 ScheduleRunTimes[0].Minute % 30 != 0 ||
                 ScheduleRunTimes[0].Second != 0 ||
                 ScheduleRunTimes[0].Millisecond != 0)
-                {
+                {
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices.Backup.Models/CommonModels/PolicyScheduleObjects.cs:119]:
```suggestion
                {
```

File: src/RecoveryServices/RecoveryServices.Backup.Models/CommonModels/PolicyScheduleObjects.cs [hunk 4]
```diff
@@ -116,12 +116,12 @@ public override void Validate()
...
                 ScheduleRunTimes[0].Second != 0 ||
                 ScheduleRunTimes[0].Millisecond != 0)
-                {
+                {   
                     throw new ArgumentException(Resources.InvalidScheduleTimeInScheduleException);
                 }
 
                 if (ScheduleRunTimes[0].Kind != DateTimeKind.Utc)
-                {
+                {
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices.Backup.Models/CommonModels/PolicyScheduleObjects.cs:124]:
```suggestion
                {
```

File: src/RecoveryServices/RecoveryServices.Backup.Providers/Providers/AzureWorkloadPsBackupProvider.cs [hunk 5]
```diff
@@ -744,7 +746,7 @@ private RestAzureNS.AzureOperationResponse<ProtectionPolicyResource> CreateorMod
                 // validate RetentionPolicy
                 AzureWorkloadProviderHelper.ValidateSQLRetentionPolicy(retentionPolicy);
                 Logger.Instance.WriteDebug("Validation of Retention policy is successful");
-
+
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices.Backup.Providers/Providers/AzureWorkloadPsBackupProvider.cs:749]:
```suggestion

```

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/RecoveryServices/RecoveryServices.Backup.Helpers/Conversions/ConversionHelpers.cs (hunk 1 above)
Comment (hiaga) [reply src/RecoveryServices/RecoveryServices.Backup.Helpers/Conversions/ConversionHelpers.cs]: Done

File: src/RecoveryServices/RecoveryServices.Backup.Helpers/Conversions/SchedulePolicyConversions.cs (hunk 2 above)
Comment (hiaga) [reply src/RecoveryServices/RecoveryServices.Backup.Helpers/Conversions/SchedulePolicyConversions.cs]: done

File: src/RecoveryServices/RecoveryServices.Backup.Models/CommonModels/PolicyScheduleObjects.cs (hunk 4 above)
Comment (hiaga) [reply src/RecoveryServices/RecoveryServices.Backup.Models/CommonModels/PolicyScheduleObjects.cs]: Done

**Review by msJinLei — APPROVED** [review]

**Pull request merged**